analysis/02_new_model_input/06.0_ghs_rural_urban_pop.md

```diff
@@ -0,0 +1,280 @@
+```
+
+```python
+grid.total_bounds
+```
+
+```python
+# converting crs and clipping
+# checking if crs are the same
+smod_raster.crs == grid.crs
+```
+
+```python
+smod_raster = rxr.open_rasterio(input_dir / "SMOD" / file_name[0])
+smod_raster.rio.crs
+```
+
+```python
+smod_raster_wgs84 = smod_raster.rio.reproject(grid.crs)
+smod_raster_wgs84_clip = smod_raster_wgs84.rio.clip_box(*grid.total_bounds)
+```
+
+```python
+smod_grid_vals = pd.DataFrame(
+    {
+        "id": grid["id"],
+        "Centroid": grid["Centroid"],
+        "urban": None,
+        "rural": None,
+        "water": None,
+    }
+)
+for grd in grid.Centroid:
+    grd_sel = grid[grid.Centroid == grd]
+    grid_rast = smod_raster_wgs84_clip.rio.clip(
+        grd_sel["geometry"], all_touched=True
+    )
+    smod_grid_vals.loc[grd_sel.index.values, ["urban"]] = (
+        (grid_rast >= 21) & (grid_rast <= 30)
+    ).sum().values / grid_rast.count().values
+    smod_grid_vals.loc[grd_sel.index.values, ["rural"]] = (
+        (grid_rast >= 11) & (grid_rast <= 13)
+    ).sum().values / grid_rast.count().values
+    smod_grid_vals.loc[grd_sel.index.values, ["water"]] = (
+        grid_rast == 10
+    ).sum().values / grid_rast.count().values
+```
+
+```python
+smod_grid_vals.tail(10)
+```
+
+```python
+del (
+    smod_raster,
+    smod_array,
+    smod_raster_wgs84,
+    smod_raster_wgs84_clip,
+)
+```
+
+## Population
+
+### Total Population by grid
+
+```python
+# downloading the popoulation data
+# selected from here: https://ghsl.jrc.ec.europa.eu/download.php?ds=pop
+phl_boxes = ["R7_C30", "R7_C31", "R8_C30", "R8_C31", "R9_C30", "R9_C31"]
+file_list = [
+    "GHS_POP_P2025_GLOBE_R2022A_54009_100_V1_0_" + patt + ".zip"
+    for patt in phl_boxes
+]
+```
+
+```python
+for file in file_list:
+    req = requests.get(pop_link + file, allow_redirects=True)
+    with zipfile.ZipFile(io.BytesIO(req.content)) as zObj:
+        fileNames = zObj.namelist()
+        for fileName in fileNames:
+            if fileName.endswith("tif"):
+                content = zObj.open(fileName).read()
+                open(input_dir / "POP" / fileName, "wb").write(content)
+```
+
+This section took long to process.
+
+NOTE:
+
+- The raster layers and the grid do not have the same CRS.
+- Options are to project grid to raster CRS or raster to grid CRS.
+- Projecting to grid CRS seems to produce realistic numbers.
+- Numbers still seem slightly distorted based on visual inspection on QGIS.
+- Example: grid `117.0E_8.2N`
+- shows a sum of 197.34 visually in QGIS
+- shows a sum of 197.34 after re-projecting the grid to raster CRS.
+- shows a sum of 201.343014 when re-projecting the rasters individually.
+- shows a sum of 355.782260 when merging and re-projecting
+- re-projecting the grid seems the way to go.
+- Merging raster layers into one takes a lot of storage.
+- It also results in higher values.
+- A re-projected raster layer takes even more space
+and is better to just re-project on the fly.
```

Comment:
Great that you noted all this down here. Could you summarize what the final set of steps look like?